src/modules/TodoCollection.js

```diff
@@ -0,0 +1,66 @@
+import { getDataFromLocalStorage } from './local-storage.js';
+import Todo from './Todo.js';
+
+export default class TodoCollection {
+  constructor(todos) {
+    this.todos = todos;
+  }
+
+  addTodo(description) {
+    const index = this.todos.length + 1;
+    const todo = new Todo({ description, index });
+    this.todos.push(todo);
+    this.setToLocalStorage();
+  }
+
+  removeTodo(index) {
+    this.todos = this.todos.filter((todo, todoIndex) => todoIndex !== index);
+    this.setToLocalStorage();
+    this.resetIndex();
+  }
+
+  setToLocalStorage() {
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  getTodos() {
+    return this.todos.sort((a, b) => b.index - a.index);
+  }
+
+  resetIndex() {
+    this.todos.forEach((todo, index) => {
+      todo.index = index + 1;
+    });
+    this.setToLocalStorage();
+  }
+
+  clearCompleted() {
+    this.todos = this.todos.filter((todo) => todo.completed !== true);
+    this.setToLocalStorage();
+    this.resetIndex();
+  }
+
+  markAsCompleted(index) {
+    for (let i = 0; i < this.todos.length; i += 1) {
+      if (this.todos[i].index === index) {
+        this.todos[i].completed = !this.todos[i].completed;
+        this.setToLocalStorage();
+        break;
+      }
+    }
+  }
```

Comment:
1. You did a great job so far 🙌🏽, and the `Adding` and `Removing` functionality is working just fine ✔, but the `Editing` and `MarkinAsCompleted` functionalities aren't working as expected 🤔.

2. When I mark a task as completed, it's not setting the `completed` property of the correct task and that is causing it if you remove that task to remove the wrong task. This also applies to editing and when you edit a task it's not editing the correct task.

- The order and index of the items in the localStorage is just fine, but the problem that is causing this bug is coming from your local array 📥.

### The Todo List
![image](https://user-images.githubusercontent.com/43172164/206740771-79ef27e1-a65c-4c17-a198-6554de90104e.png)

### Your local storage at the same time
![image](https://user-images.githubusercontent.com/43172164/206740896-f17bdd92-699e-42e2-8044-9c453edd0462.png)